File: src/InfrastructurePlugins/BaseModule/src/Application/Dtos/GridSearchColumnFilter.cs
using InfrastructurePlugins.BaseModule.Components.QueryBuilder;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace InfrastructurePlugins.BaseModule.Application.Dtos
{
    [Description("列过滤信息")]
    public class GridSearchColumnFilter
    {
        [Description("列名")]
        public string Column { get; set; }
        [Description("匹配模式")]
        public GridSearchColumnFilterMatchMode MatchMode { get; set; }
        [Description("过滤值")]
        public object Value { get; set; }

        [Description("子表达式")]
        public bool IsChildExpress { get; set; }

        [Description("属性类型")]
        public Type ProperyType { get; set; }

        [Description("正则表达式")]
        public string RegExp { get; set; }

        [Description("属性名称")]
        public string PropertyName { get; set; }

        [Description("操作符")]
        public OpertionSymbol OpertionSymbol { get; set; }

        [Description("值1")]
        public object Value1 { get; set; }

        [Description("值2")]
        public object Value2 { get; set; }

        [Description("逻辑连接符")]
        public ConcatType Concat { get; set; }
        [Description("子表达式查询条件")]
        public List<GridSearchColumnFilter> Childs { get; set; } = new List<GridSearchColumnFilter>();
    }
}

File: src/FrontEndPlugins/MVVM.Angular.Website/static/src/modules/generated_module/dtos/grid-search-column-filter.ts
import { GridSearchColumnFilterMatchMode } from './grid-search-column-filter-match-mode';
import { OpertionSymbol } from './opertion-symbol';
import { SetOpertionSymbol } from './set-opertion-symbol';
import { ConcatType } from './concat-type';

/** 列过滤信息 */
export class GridSearchColumnFilter {
    /** 列名 */
    public Column: string;
    /** 匹配模式 */
    public MatchMode: GridSearchColumnFilterMatchMode;
    /** 过滤值 */
    public Value: any;
    /** 子表达式 */
    public IsChildExpress: boolean;
    /** 子查询 */
    public IsChildQuery: boolean;
    /** 属性类型 */
    public PropertyType: any;
    /** 正则表达式 */
    public RegExp: string;
    /** 属性名称 */
    public PropertyName: string;
    /** 操作符 */
    public OpertionSymbol: OpertionSymbol;
    /** 集合操作符 */
    public SetOpertionSymbol: SetOpertionSymbol;
    /** 值1 */
    public Value1: any;
    /** 值2 */
    public Value2: any;
    /** 逻辑连接符 */
    public Concat: ConcatType;
    /** 子表达式查询条件 */
    public Childs: GridSearchColumnFilter[];
}
